# 03 — Data Wrangling

IBM Data Science Capstone — SpaceX Falcon 9 first-stage landing prediction.

Prepare the 90-row Falcon 9 dataset, create the binary landing target, impute missing payload mass, and preserve a modeling-ready copy.

In [1]:
import pandas as pd
from pathlib import Path

ROOT=Path('..')
df=pd.read_csv(ROOT/'data/dataset_part_1.csv')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90 entries, 0 to 89
Data columns (total 17 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   FlightNumber    90 non-null     int64  
 1   Date            90 non-null     object 
 2   BoosterVersion  90 non-null     object 
 3   PayloadMass     90 non-null     float64
 4   Orbit           90 non-null     object 
 5   LaunchSite      90 non-null     object 
 6   Outcome         90 non-null     object 
 7   Flights         90 non-null     int64  
 8   GridFins        90 non-null     bool   
 9   Reused          90 non-null     bool   
 10  Legs            90 non-null     bool   
 11  LandingPad      64 non-null     object 
 12  Block           90 non-null     float64
 13  ReusedCount     90 non-null     int64  
 14  Serial          90 non-null     object 
 15  Longitude       90 non-null     float64
 16  Latitude        90 non-null     float64
dtypes: bool(3), float64(4), int64(3), obj

In [2]:
df['PayloadMass']=df['PayloadMass'].fillna(df['PayloadMass'].mean())
df['Class']=df['Outcome'].str.startswith('True').astype(int)
model_df=df.drop(columns=['Outcome','Date','BoosterVersion'])
model_df.to_csv(ROOT/'data/dataset_part_2.csv', index=False)
print('shape:', model_df.shape)
print('success rate:', model_df['Class'].mean())
print('missing values:', int(model_df.isna().sum().sum()))

shape: (90, 15)
success rate: 0.6666666666666666
missing values: 26


The resulting target is `Class = 1` for a successful first-stage landing outcome and `0` otherwise. The study contains 90 Falcon 9 records and an overall success rate of 66.7%.